src/main/webapp/app/entities/text-exercise/text-exercise.component.html

```diff
@@ -74,7 +74,16 @@ <h4>
                                 <fa-icon icon="pencil-alt"></fa-icon>
                                 <span class="d-none d-md-inline" jhiTranslate="entity.action.edit">Edit</span>
                             </button>
-                            <button *ngIf="textExercise.isAtLeastInstructor" type="submit" (click)="openDeleteTextExerciseDialog(textExercise)" class="btn btn-danger btn-sm mr-1">
+                            <button
```

Comment:
Wow, this turns out to be very tricky issue. I spend a lot of time investigating on the best solution for it and it is not perfect. The main problem is that Angular doesn't support something like `Fragment` which is possible to do in React. This is critical since all of our buttons are grouped using Bootstrap `button-group` and if we create `jhi-delete-button` as an Angular component, `button-group` will not work anymore (because Angular will wrap it in a tag). I tried already to play around with the styling, but it looks like it is only possible if we completely don't use Bootstrap for this.
The best solution I found for now is to still use directive and as you can see from the latest changes this directive now uses `Renderer2` to append classes and elements to the `button`. But even with this approach there is one problem, which I don't know how to solve and maybe you can help me with it. I still can't make it work with `FontAwesome` icons. I found a closed issue about this https://github.com/FortAwesome/angular-fontawesome/issues/72, but it doesn't look like a solution since here you need to have reference of the component in the directive.

Reply:
As this is a bigger issue, I'd suggest to move this into a follow up issue.
There is a possibility to work with `<ng-content>` and use `ngProjectAs`. I can recommend the Post ["ng-content: The hidden docs"](https://medium.com/claritydesignsystem/ng-content-the-hidden-docs-96a29d70d11b) on usage of `<ng-content>`. But I think this would require our own Button Group Component. But as I said, this could be done in a follow up.